### ISSUE TITLE
Rationale for supporting `application/x-www-form-urlencoded` response

### ISSUE THREAD

**Issue by ramosbugs** [alexcrichton/oauth2-rs#33]
It looks like the `request_token` function supports both a JSON response and an `application/x-www-form-urlencoded` response. From my reading of the spec (https://tools.ietf.org/html/rfc6749), response bodies are always JSON:
> The parameters are included in the entity-body of the HTTP response
   using the "application/json" media type as defined by [RFC4627].  The
   parameters are serialized into a JavaScript Object Notation (JSON)
   structure by adding each parameter at the highest structure level.
   Parameter names and string values are included as JSON strings.
   Numerical values are included as JSON numbers.  The order of
   parameters does not matter and can vary.

It looks like OAuth 1.0 (https://tools.ietf.org/html/rfc5849) used `application/x-www-form-urlencoded` responses. Since this library only supports OAuth 2.0, do we need to support this response type? Technically, that's a deviation from the spec.

It might be worth removing this in the next major version since it'll break backward compatibility if anyone happens to be relying on this behavior. I'll be proposing a change soon (for #27) that would require the major version bump, so we might as well lump this in unless there's a good reason to keep it.

**Comment by ramosbugs** [comment]
It looks like this was due to GitHub's early implementation of OAuth2, before the spec was finalized (https://twitter.com/patricktoomey/status/976873772601966593). As long as the client adds the `Accept: application/json` header, it should be safe to remove support for parsing form-encoded token responses. That will simplify the code significantly. Planning to roll this into upcoming 2.0.0 breaking changes. Let me know of anyone objects.

**Comment by SorenHolstHansen** [comment]
Hi, sorry for using an old issue, but Slack is using application/x-www-form-urlencoded for their code exchange. I am for the moment just doing the code exchange request myself with reqwest, but would it be possible to support this again?

**Comment by ramosbugs** [comment]
Hey @SorenHolstHansen,

Would you mind linking me to the relevant Slack docs? I couldn't find anything that said they require `application/x-www-form-urlencoded`.

As a workaround, I'd suggest writing a shim function to pass to the `request`/`request_async` methods that converts the `application/x-www-form-urlencoded` response to a JSON one before passing the `HttpResponse` back to this crate. There doesn't seem to be wide enough usage or anything in the published standards to justify adding this functionality back to the crate itself.

Hope that helps!

**Comment by SorenHolstHansen** [comment]
Yeah, it is not easy to find, but you can see it in their [api reference](https://api.slack.com/methods/oauth.v2.access).

It is fine by me that it isn't supported, though, the workaround is simple enough.
